File: 141-linked-list-cycle/141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head is None:
            return False
        ptr = head
        ptr2 = head.next
        while ((ptr is not None) & (ptr2 is not None) & (ptr != ptr2)):
            try:
                ptr2 = ptr2.next.next
                ptr = ptr.next
            except:
                return False
        if ptr == ptr2:
            return True
        return False
            

            

File: 141-linked-list-cycle/141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if (head==null) {
            return false;
        }
        ListNode ptr = head;
        ListNode ptr2 = head.next;
        boolean flag = true;
        while (ptr2 != null && ptr != null && ptr != ptr2) {
            if (!flag) {
                ptr = ptr.next;
                ptr2 = ptr2.next;
                flag = true;
            }
            else {
                ptr2 = ptr2.next;
                flag = false;
            }
        }
        if (ptr == ptr2) {
            return true;
        }
        return false;
    }
}